modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -0,0 +1,217 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/gapi/util/throw.hpp> // throw_error
+#include <opencv2/gapi/streaming/format.hpp> // kernels
+
+#include "api/gbackend_priv.hpp"
+#include "backends/common/gbackend.hpp"
+
+#include "gstreamingbackend.hpp"
+#include "gstreamingkernel.hpp"
+
+namespace {
+
+struct StreamingCreateFunction
+{
+    static const char *name() { return "StreamingCreateFunction";  }
+    cv::gapi::streaming::CreateActorFunction createActorFunction;
+};
+
+using StreamingGraph = ade::TypedGraph
+    < cv::gimpl::Op
+    , StreamingCreateFunction
+    >;
+
+using ConstStreamingGraph = ade::ConstTypedGraph
+    < cv::gimpl::Op
+    , StreamingCreateFunction
+    >;
+
+
+class GStreamingIntrinExecutable final: public cv::gimpl::GIslandExecutable
+{
+    virtual void run(std::vector<InObj>  &&,
+                     std::vector<OutObj> &&) override {
+        GAPI_Assert(false && "Not implemented");
+    }
+
+    virtual void run(GIslandExecutable::IInput &in,
+                     GIslandExecutable::IOutput &out) override;
+
+    virtual bool allocatesOutputs() const override { return true; }
+    virtual cv::RMat allocate(const cv::GMatDesc&) const override { return {}; }
+
+    virtual bool canReshape() const override { return true; }
+    virtual void reshape(ade::Graph& graph, const cv::GCompileArgs&) override {
+        // The whole graph here consists only of 1 operation with 1 input node and 1 output one
+        cv::gimpl::GModel::ConstGraph new_gm(graph);
+
+        for (const auto& node_new : graph.nodes())
+        {
+            if (new_gm.metadata(node_new).get<cv::gimpl::NodeType>().t == cv::gimpl::NodeType::DATA) {
+                auto data_new = new_gm.metadata(node_new).get<cv::gimpl::Data>();
+                for (auto&& node : m_g.nodes()) {
+                    if (m_gm.metadata(node).get<cv::gimpl::NodeType>().t == cv::gimpl::NodeType::DATA) {
+                        auto data_old = m_gm.metadata(node).get<cv::gimpl::Data>();
+                        if (data_new.rc == data_old.rc && data_new.shape == data_old.shape) {
+                            data_old.meta = data_new.meta;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
No.

Metadata propagation happens automatically via generic compiler passes. You don't need to do this here.

What this method is supposed to do in backends is -- if a backend allocates some memory based on metadata, e.g. for intermediate/island-internal buffers, you just reallocate those buffers to new formats. Or, if your backend (like fluid) optimizes execution based on image geometry, you regenerate your internal state to new image sizes.

In your backend, which does nothing, you can leave this method empty.

Reply:
Also (fyi) - `new_gm` and `m_gm` point to the same graph since due to reshape, the original `ade::Graph` object is preserved.

Reply:
Thanks for explanation, removed